### ISSUE TITLE
Choice values in OPTIONS calls are always represented as strings, regardless of type

### ISSUE THREAD

**Issue by unreality** [tomchristie/django-rest-framework#4111]
If you call OPTIONS on a model that has fields with choices, the displayed choices are always strings eg:

```
"account": {
    "type": "field",
    "required": true,
    "read_only": false,
    "label": "Account",
     "choices": [
        {
            "display_name": "Choice 1 Display Name",
            "value": "71518788"
        },
        {
            "display_name": "Choice 2 Display Name",
            "value": "71519217"
        }
        ....etc
```

Currently for `RelatedField`, when you request the `choices` property, all values are converted to strings, but this probably doesn't make sense on things such as `PrimaryKeyRelatedField`, or other field values which are integers or booleans.

**Comment by bdewitt1991** [comment]
Black list string

**Comment by ansumanbebarta** [comment]
May I have steps to reproduce this? I am interested to look into it. I tried to reproduce but I couldn't.

**Comment by bdewitt1991** [comment]
Open

On Jun 12, 2016 5:41 PM, "Ansuman Bebarta" <notifications@github.com> wrote:
>
> May I have steps to reproduce this? I am interested to look into it. I
tried to reproduce but I couldn't.

**Comment by unreality** [comment]
Here is an example:


```
class ExampleTypeModel(models.Model):
    id = models.AutoField(primary_key=True)
    label = models.CharField(max_length=100)
    def __str__(self):
        return self.label

class ExampleModel(models.Model):
    id = models.AutoField(primary_key=True)
    example_type = models.ForeignKey('ExampleTypeModel', models.DO_NOTHING)
    name = models.CharField(max_length=100)
```

```
class ExampleModelSerializer(serializers.ModelSerializer):
    class Meta:
        model = ExampleModel

class ExampleTypeModelSerializer(serializers.ModelSerializer):
    class Meta:
        model = ExampleTypeModel
```

```
class ExampleModelView(viewsets.ModelViewSet):
    serializer_class = ExampleModelSerializer
    queryset = ExampleModel.objects.all()
```

If you add the viewset to your url router, and request OPTIONS, the choices for the foreign key (ExampleTypeModel) will all be strings.

**Comment by ansumanbebarta** [comment]
Hi,

Thank you for the steps you have provided. I am getting following while requesting OPTIONS. But I am not getting choices for the foreign key.

```json
{
    "name": "Example Model List",
    "description": "",
    "renders": [
        "application/json",
        "text/html"
    ],
    "parses": [
        "application/json",
        "application/x-www-form-urlencoded",
        "multipart/form-data"
    ],
    "actions": {
        "POST": {
            "id": {
                "type": "integer",
                "required": false,
                "read_only": true,
                "label": "Id"
            },
            "name": {
                "type": "string",
                "required": true,
                "read_only": false,
                "label": "Name",
                "max_length": 100
            },
            "example_type": {
                "type": "field",
                "required": true,
                "read_only": false,
                "label": "Example type"
            }
        }
    }
}
```

I have forked the lastest django-rest-framework repo so that I can create PR. Now after a bit of searching I found following:
https://github.com/tomchristie/django-rest-framework/pull/4021
Seems like we don't show choices anymore for related fields. So is it a BUG anymore?

**Comment by tomchristie** [comment]
Closed, both via #4021 (Don't show in OPTIONS - thanks @ansumanbebarta!) and via #4379 (.choices property of RelatedField should preserve non-string values.)
